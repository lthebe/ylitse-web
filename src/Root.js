import React from 'react';
import { Switch, Redirect, Route } from 'react-router-dom';

import Navigation from './Navigation';
import MissingPage from './MissingPage';
import SkillsPage from './SkillsPage';
import ProfileSheet from './ProfileSheet';

const Root = () => (
    <main>
        <Route component={Navigation} />
        <Switch>
            <Route path="/" exact><Redirect to="/accounts" /></Route>
            <Route path="/accounts" exact component={ProfileSheet} />
            <Route path="/skills" exact component={SkillsPage} />
            <Route component={MissingPage} />
        </Switch>
    </main>
);

export default Root;
