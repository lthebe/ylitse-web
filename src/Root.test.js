import React from 'react';
import { createShallow } from 'material-ui/test-utils';
import Root from './Root';

describe('Root', () => {
    let shallow;
    beforeAll(() => {
        shallow = createShallow();
    });
    afterAll(() => {
        shallow.cleanUp();
    });
    test('renders correctly', () => {
        // ProfileSheet is a HOC, so diving
        const shallowWrapper = shallow(<Root />).dive();
        expect(shallowWrapper).toMatchSnapshot();
    });
});
