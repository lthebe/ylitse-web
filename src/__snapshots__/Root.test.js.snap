// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Root />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
        />,
        <Switch>
          <Route
            exact={true}
            path="/"
          >
            <Redirect
              push={false}
              to="/accounts"
            />
          </Route>
          <Route
            component={[Function]}
            exact={true}
            path="/accounts"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/skills"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
            >
              <Redirect
                push={false}
                to="/accounts"
              />
            </Route>,
            <Route
              component={[Function]}
              exact={true}
              path="/accounts"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/skills"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Redirect
                push={false}
                to="/accounts"
              />,
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "push": false,
                "to": "/accounts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/accounts",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/skills",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
          />,
          <Switch>
            <Route
              exact={true}
              path="/"
            >
              <Redirect
                push={false}
                to="/accounts"
              />
            </Route>
            <Route
              component={[Function]}
              exact={true}
              path="/accounts"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/skills"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
              >
                <Redirect
                  push={false}
                  to="/accounts"
                />
              </Route>,
              <Route
                component={[Function]}
                exact={true}
                path="/accounts"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/skills"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Redirect
                  push={false}
                  to="/accounts"
                />,
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "push": false,
                  "to": "/accounts",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/accounts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/skills",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
